c/driver/common/utils.c

```diff
@@ -1007,7 +1007,8 @@ struct AdbcGetObjectsCatalog* AdbcGetObjectsDataGetCatalogByName(
     for (int64_t i = 0; i < get_objects_data->n_catalogs; i++) {
       struct AdbcGetObjectsCatalog* catalog = get_objects_data->catalogs[i];
       struct ArrowStringView name = catalog->catalog_name;
-      if (!strncmp(name.data, catalog_name, name.size_bytes)) {
+      if ((int64_t)strlen(catalog_name) &&
```

Comment:
this check seems to be missing its RHS?

Reply:
you could factor out a `StringViewEquals(struct ArrowStringView, char*)` for these checks to make it clearer if you want